### PR TITLE
Implemented Genetics in blueberry bush.

### PR DESCRIPTION
**Pull request by arpan98**
Feature for #46 .

Relies on the changes made in the Genome module in this PR: https://github.com/Terasology/Genome/pull/7.

Implemented a genetic system including mutation and breeding from the Genome module into SimpleFarming. Currently implemented only for blueberry, varying the Food filling component.

Might be a little difficult to test due to only one character gene (mutation is the only thing that matters), so try increasing the mutation rate.

Things to do further:
- Extend to all bushes, vines, trees.
- Change the look (tiles) of the Seed Mixer. Currently same as a chest.
- Discuss and decide on a gene to property mapping that allows for effective breeding.
- Should the user be given some feedback as to how the breeding went? In my opinion, no, to keep it natural.

### CODE REVIEW

**Comment by Cervator** [comment]
Checked it out real quick and it looks fine, although it is indeed a bit hard to test. Maybe you can think of a debug console command or two to help it out, or some sort of seed examiner item?

It would also help if the UI had a bit more of an indication of what it does. Rather than 4 boxes in a row try to add some minimal decoration so maybe it ends up looking like:

[] + [] -> [][]

**Comment by arpan98** [comment]
I tried to implement your second point, but i could not find a way to. I looked at the Workstation module and it is done there dynamically by adding UIWidgets in Java. Is there a way to do it static?

**Comment by Cervator** [comment]
@xrtariq2594 mind doing a bit of general testing for SimpleFarming then merge this? I can confirm the seed mixer works, but didn't test anything in general (like if seeds with or without mixing actually grow, etc)

**Comment by mjuvekar7** [comment]
Maybe add a second variant of the blueberry with a different genome? Would make testing a _ton_ easier.

**Comment by mjuvekar7** [comment]
I spoke with @arpan98 and he came up with a general testing guide for this PR. Thought it might be helpful if I sent it here.

1. `give seedmixer 1`, place it, fiddle with it.
2. try `give blueberryseed`, should give you a blueberry seed with genome F (default). You can use the command `heldGenomeCheck` to check genome of held item.
3. Drag blueberries onto the input slots of the seedmixer- only seeds should be able to be dropped on the slots.
4. Check genome of input and output seeds. If you want, you can fiddle around with the mutation constant in the GenomeSystem to see what works best (I just put in a random value for now).
5. Try growing the seeds, checking genome of resultant blueberry.

File: src/main/java/org/terasology/simpleFarming/systems/GeneticsSystem.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @In
+    private GenomeRegistry genomeRegistry;
+
+    @Override
+    public void preBegin() {
+        int genomeLength = 1;
+
+        //A=5, F=10, K=15
+        GeneMutator geneMutator = new VocabularyGeneMutator("ABCDEFGHIJK");
+        BreedingAlgorithm breedingAlgorithm = new MonoploidBreedingAlgorithm(0, 0.05f, geneMutator);
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/simpleFarming/systems/GeneticsSystem.java]: I would suggest you make the mutation chance (the 0.05f parameter here) a separate variable. That would increase readability and would also make it easier to tweak.

**Comment by mjuvekar7** [comment]
Maybe add a second variant of the blueberry with a different genome? Would make testing a lot easier.

**Comment by mjuvekar7** [comment]
Hey, so because of the recent refactor to this module, I think we might need to rework this whole PR. Adding more commits onto this PR will just make it messy, so I think we should close this one and create a new one. We should still be able to salvage large portions of the code.

Pinging @arpan98 @Cervator

**Comment by mjuvekar7** [comment]
I also think that the Hunger stuff isn't the feature we would want to add first, as it involves adding a dependency to Hunger, which we may or may not want to do (ping @jellysnake ). I would say we should start off with things like flower/bush color or something simple with two possible varieties (one dominant, one recessive).

**Comment by skaldarnar** [comment]
Sorry for coming back to this only now 🙈  I'm closing this due to date and conflicts.

**Pull request closed without merging**